modules/library/main/src/main/java/org/geotools/filter/Filters.java

```diff
@@ -412,6 +414,17 @@ public static double number(Object value) {
                 throw new IllegalArgumentException("Unable to decode '" + text + "' as a number");
             }
         }
+        // if value is coming from ComplexAttribute through Xpath
+        if (value instanceof ComplexAttribute) {
```

Comment:
Why does it even get there, the calling code should evaluate the expression/attribute before getting here.

Reply:
@aaime thanks for the review, so I am guessing it should be processed outside number method, where we can ensure whatever gets into this method will always be a number. Is that acceptable.?